catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/TaskRepository.java

```diff
@@ -0,0 +1,359 @@
+
+    // Update owner relationship
+    setFields(storedDB, TASK_UPDATE_FIELDS); // First get the ownership information
+    updateOwner(storedDB, storedDB.getOwner(), newOwner);
+
+    // Service can't be changed in update since service name is part of FQN and
+    // change to a different service will result in a different FQN and creation of a new task under the new service
+    storedDB.setService(service);
+    applyTags(updatedTask);
+
+    return new PutResponse<>(Status.OK, storedDB);
+  }
+
+  @Transaction
+  public Task patch(String id, JsonPatch patch) throws IOException {
+    Task original = setFields(validateTask(id), TASK_PATCH_FIELDS);
+    Task updated = JsonUtils.applyPatch(original, patch, Task.class);
+    patch(original, updated);
+    return updated;
+  }
+
+  public Task createInternal(Task task, EntityReference service, EntityReference owner) throws IOException {
+    task.setFullyQualifiedName(getFQN(service, task));
+    EntityUtil.populateOwner(userDAO(), teamDAO(), owner); // Validate owner
+
+    // Query 1 - insert task into task_entity table
+    taskDAO().insert(JsonUtils.pojoToJson(task));
+    setService(task, service);
+    setOwner(task, owner);
+    applyTags(task);
+    return task;
+  }
+
+  private void applyTags(Task task) throws IOException {
+    // Add task level tags by adding tag to task relationship
+    EntityUtil.applyTags(tagDAO(), task.getTags(), task.getFullyQualifiedName());
+    task.setTags(getTags(task.getFullyQualifiedName())); // Update tag to handle additional derived tags
+  }
+
+  private void patch(Task original, Task updated) throws IOException {
+    String taskId = original.getId().toString();
+    if (!original.getId().equals(updated.getId())) {
+      throw new IllegalArgumentException(CatalogExceptionMessage.readOnlyAttribute(Entity.TASK, "id"));
+    }
+    if (!original.getName().equals(updated.getName())) {
+      throw new IllegalArgumentException(CatalogExceptionMessage.readOnlyAttribute(Entity.TASK, "name"));
+    }
+    if (updated.getService() == null || !original.getService().getId().equals(updated.getService().getId())) {
+      throw new IllegalArgumentException(CatalogExceptionMessage.readOnlyAttribute(Entity.TASK, "service"));
+    }
+    // Validate new owner
+    EntityReference newOwner = EntityUtil.populateOwner(userDAO(), teamDAO(), updated.getOwner());
+
+    EntityReference newService = updated.getService();
+    // Remove previous tags. Merge tags from the update and the existing tags
+    EntityUtil.removeTags(tagDAO(), original.getFullyQualifiedName());
+    updated.setHref(null);
+    updated.setOwner(null);
+    updated.setService(null);
+    taskDAO().update(taskId, JsonUtils.pojoToJson(updated));
+    updateOwner(updated, original.getOwner(), newOwner);
+    updated.setService(newService);
+    applyTags(updated);
+  }
+
+  public EntityReference getOwner(Task task) throws IOException {
+    if (task == null) {
```

Comment:
nit. Can be turned into a single statement

Reply:
Done